shell/platform/common/alert_platform_node_delegate.h

```diff
@@ -0,0 +1,41 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_COMMON_ALERT_PLATFORM_NODE_DELEGATE_H_
+#define FLUTTER_SHELL_PLATFORM_COMMON_ALERT_PLATFORM_NODE_DELEGATE_H_
+
+#include "flutter/third_party/accessibility/ax/ax_node_data.h"
+#include "flutter/third_party/accessibility/ax/platform/ax_platform_node_delegate_base.h"
+
+namespace flutter {
+
+class AlertPlatformNodeDelegate : public ui::AXPlatformNodeDelegateBase {
+ public:
+  explicit AlertPlatformNodeDelegate(
+      ui::AXPlatformNodeDelegate* parent_delegate);
```

Comment:
The parent delegate should never be `nullptr`, right? Consider making this a `const ui::AXPlatformNodeDelegate&` instead and updating the `AlertPlatformMessage` test to pass a mocked object instead.

Reply:
I do not think we can make this `const` without a more thorough sweeping change to `AXPlatformNodeDelegate` and classes that touch it. `AlertPlatformNodeDelegate` needs to call several methods on its `parent_delegate_` that are not `const`.

Reply:
Ah ok. My meta point is to disallow `nullptr`, could we make it a non-constant reference?

Reply:
We should be able to.